v2/svid/jwtsvid/svid_test.go

```diff
@@ -0,0 +1,521 @@
+package jwtsvid_test
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/jwtbundle"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/svid/jwtsvid"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/cryptosigner"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+const hs256Token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG" +
+	"4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
+
+func TestParseAndValidate(t *testing.T) {
+	// Create numeric dates
+	issuedAt := jwt.NewNumericDate(time.Now())
+	expiresTime := time.Now().Add(time.Minute)
+	expires := jwt.NewNumericDate(expiresTime)
+
+	// Create keys
+	key, err := ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
```

Comment:
Key generation time can add up across lots of test cases. I'm ok with you leaving these as-is for now. I'll be adding some small test scaffolding to use pregenerated keys to keep things speedy.

Reply:
moved to a package level var